integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/preaggregate/TestPreAggregateLoad.scala

```diff
@@ -310,5 +310,99 @@ test("check load and select for avg double datatype") {
     checkAnswer(sql("select name,avg(salary) from maintbl group by name"), rows)
   }
 
+  test("create datamap with 'if not exists' after load data into mainTable and create datamap") {
```

Comment:
I think no need to add test cases in all the files. One test case in TestPreAggregateCreateCommand and one in TestTimeseriesCreateTable would be enough.

Reply:
not all file, but I think we should add this test case:  Create data map when the main table has data.

Reply:
And another problem happend when the order of the load data into main table and create datamap is different, you can check CARBONDATA-2085.